File: src/problems_all/1588_sum_of_all_odd_length_subarrays/1588_sum_of_all_odd_length_subarrays.py
"""
https://leetcode.com/problems/sum-of-all-odd-length-subarrays/

Runtime: 89 ms
Memory: 14.9 MB
https://leetcode.com/problems/sum-of-all-odd-length-subarrays/submissions/877513125/
"""


from typing import List


class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        odd_nums = []
        for i in range(1, len(arr) + 1, 2):
            odd_nums.append(i)

        odd_arrays = []
        for i in odd_nums:
            for j in range(len(arr)):
                if len(arr[j:j+i]) >= i:
                    odd_arrays += arr[j:j+i]

        return sum(odd_arrays)


def main():
    arr = [1, 2]
    result = Solution().sumOddLengthSubarrays(arr)
    print(result)


if __name__ == '__main__':
    main()

File: src/problems_all/1588_sum_of_all_odd_length_subarrays/1588_sum_of_all_odd_length_subarrays.go
/*
https://leetcode.com/problems/sum-of-all-odd-length-subarrays/

Runtime: 9 ms
Memory: 7.7 MB
https://leetcode.com/problems/sum-of-all-odd-length-subarrays/submissions/877522776/
*/

package main

import "fmt"

func main() {
	arr := []int{10, 11, 12}
	result := sumOddLengthSubarrays(arr)
	fmt.Println(result)
}

func sumOddLengthSubarrays(arr []int) int {
	oddNums := []int{}
	for i := 1; i < len(arr)+1; i += 2 {
		oddNums = append(oddNums, i)
	}

	oddArrayItems := []int{}
	for _, i := range oddNums {
		for j := 0; j < len(arr); j++ {
			if (i+j <= len(arr)) && (len(arr[j:j+i]) >= i) {
				oddArrayItems = append(oddArrayItems, arr[j:j+i]...)
			}
		}
	}

	return sum(oddArrayItems)
}

func sum(arr []int) int {
	total := 0
	for _, i := range arr {
		total += i
	}
	return total
}
